_plugins/filters/meta_util.rb

```diff
@@ -0,0 +1,86 @@
+# traverses the page and site objects to find the most appropriate
+# meta data tags based on a predetermined priority
+
+module Jekyll
+  module CRDS
+    class MetaUtil
+      # Priority:
+      # 1) meta image
+      # 2) page image
+      # 3) page background image
+      # 4) first image found in page content
+      # 5) site variable's image
+      # Don't love it -- some pretty brittle checks based on naming conventions established in media
+      # method for getting the url for the <meta name="image" content="{{ meta_image }}"> tag in the head of a SSG html page
+      def self.get_meta_image(page_meta_image, page_image, page_bg_image, page_content, site_image)
```

Comment:
Checkout [the splat operator](https://medium.freecodecamp.org/rubys-splat-and-double-splat-operators-ceb753329a78). You can use it to create catch-all arguments so you can pass in whatever you want without having predefine all the params you might need... 

```ruby
def self.get_meta_image(*args)
  puts args.inspect     # [ 'https://page-meta-image.jpg', 'https://page-image.jpg', ... ]
end

self.get_meta_image(page_meta_image, page_image, ...)
```

Reply:
@tcmacdonald I'll look to leverage the splat operator wherever possible. Is it applicable where the meta image method is concerned since I need to potentially execute a regex against the 4th argument (page_content)?

Reply:
@tcmacdonald Applied this to the get_meta_description method. Thank you!